reconcile/utils/helm.py

```diff
@@ -108,12 +108,15 @@ def template(
 def template_all(
     url: str,
     path: str,
+    ref: str,
     namespace: str,
     values: Mapping[str, Any],
     ssl_verify: bool = True,
 ) -> Iterable[Mapping[str, Any]]:
     with tempfile.TemporaryDirectory() as wd:
         git.clone(url, wd, depth=1, verify=ssl_verify)
+        git.fetch(ref, wd)
+        git.checkout(ref, wd)
```

Comment:
if we always need `fetch & checkout`, then we can skip checkout during clone to save time


```shell
git clone --no-checkout --depth 1 url
git fetch --depth 1 origin sha
git checkout sha
```

In fact, the implicit fetch during clone is wasted since we only care about one commit, the most efficient way to fetch it is:


```shell
git init repo
cd repo
git remote add origin url
git fetch --depth 1 origin sha
git checkout sha
```

Reply:
went with a conditional fetch, and also implemented fetch depth.